### ISSUE TITLE
aws_spot_fleet_request valid_until doesn't support indefinite requests

### ISSUE THREAD

**Issue by quintok** [terraform-providers/terraform-provider-aws#12755]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.12.24
+ provider.aws v2.56.0

### Affected Resource(s)

* aws_spot_fleet_request

### Terraform Configuration Files

```hcl
resource "aws_spot_fleet_request" "cluster" {
...
  valid_until = "2020-05-06T23:53:38Z"
...
}
```

### Expected Behavior

The AWS API says valid_until is an optional parameter.  Unfortunately the terraform implementation defaults to 24 hours.  I don't see a way I can configure this resource to get it to behave like the aws default behaviour of "until the spot request is cancelled" - see ValidUntil in https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotFleetRequestConfigData.html

### Actual Behavior

defaults to 24 hours, has no obvious way to set it to "until the request is cancelled"

Potential workarounds I thought of short term are:
* Set a ridiculous date in the future, not sure if the AWS API has issues with that.
* Have a null_resource call the aws api to remove the value and set that value as a ignore_changes proeprty in tf.

**Comment by ewbankkit** [comment]
@quintok Thanks for raising this.
It looks like https://github.com/terraform-providers/terraform-provider-aws/pull/9718 will address it.

**Comment by bflad** [comment]
The change to allow the `valid_until` argument to no longer default to 24 hours has been merged and will release with version 3.0.0 of the Terraform AWS Provider, likely in two weeks. Thanks to @jkburges for the implementation. 👍
